homeassistant/components/shopping_list/__init__.py

```diff
@@ -53,10 +55,23 @@
 )
 
 
-@asyncio.coroutine
-def async_setup(hass, config):
+async def async_setup(hass, config):
     """Initialize the shopping list."""
 
```

Comment:
Can you add : 
```py
if DOMAIN not in config:
        return True
```

Reply:
Then all the tests are failing and it's not working anymore

Reply:
What did you add and what tests are failing? It's easier if you push the change so we can see it.

Reply:
Tests shouldn't fail with this

Reply:
I pushed the changes, which results in the tests failing:

```
    async def test_ws_add_item_fail(hass, hass_ws_client, sl_setup):
        """Test adding shopping_list item failure websocket command."""
        client = await hass_ws_client(hass)
        await client.send_json({"id": 5, "type": "shopping_list/items/add", "name": 123})
        msg = await client.receive_json()
        assert msg["success"] is False
>       assert len(hass.data["shopping_list"].items) == 0
E       KeyError: 'shopping_list'
```

Reply:
The tests need to be updated. It's no longer enough to set up the component without config. Either set up the component with config or set up a config entry. In tests we can create a  mock config entry and add it to hass, then call `hass.config_entries.async_setup` passing the entity_id.

MockConfigEntry is here:
https://github.com/home-assistant/core/blob/007d934214f5a92db69ac3fbfe5470a54d19a459/tests/common.py#L660-L698

Reply:
Perfect, works now